components/registry/org.wso2.carbon.registry.resource.ui/src/main/java/org/wso2/carbon/registry/resource/ui/Utils.java

```diff
@@ -254,4 +254,44 @@ public static boolean hasDependencies(DependenciesBean dependenciesBean,String s
         return new String[0][];
     }
 
+    /**
+     * Adding new property to end of the properties array
+     *
+     * @param properties String[][] of properties
+     * @return propertyArray    updated String[][] of properties
+     */
+    public static String[][] setProperties(String[][] properties, String key, String value) {
+        String[][] propertyArray = new String[properties.length + 1][2];
+        for (int i = 0; i <= properties.length; i++) {
+            if (i < properties.length) {
+                propertyArray[i][0] = properties[i][0];
+                propertyArray[i][1] = properties[i][1];
+            } else {
+                propertyArray[properties.length][0] = key;
+                propertyArray[properties.length][1] = value;
+                return propertyArray;
+            }
+        }
+        return new String[0][];
+    }
+
+    /**
+     * return the concatenated full path of the resource
+     *
+     * @param parentPath
+     * @param resourceName
+     * @return resourcePath
```

Comment:
/**
     * return the concatenated full path of the resource
     *
     * @param parentPath location of the resource in registry
     * @param resourceName resource name which entered by user
     * @return resourcePath.toString() full path of the resource going to be saved
     */